src/runtime/audience-action-flow.ts

```diff
@@ -179,12 +181,25 @@ export class AudienceActionIframeFlow implements AudienceActionFlow {
   private handleCompleteAudienceActionResponse_(
     response: CompleteAudienceActionResponse
   ): void {
+    const {onResult, configurationId} = this.params_;
     this.dialogManager_.completeView(this.activityIframeView_);
     this.entitlementsManager_.clear();
     const userToken = response.getSwgUserToken();
     if (userToken) {
       this.deps_.storage().set(Constants.USER_TOKEN, userToken, true);
     }
+    if (this.isOptIn(this.params_.action) && onResult) {
```

Comment:
Out of curiosity, what's the plan to handle non-opt-in cases?

Reply:
For surveys it is handled in the data transfer request. For rewarded ads it will not be called.